networking/eth2/src/main/java/tech/pegasys/artemis/networking/eth2/peers/PeerChainValidator.java

```diff
@@ -58,28 +59,33 @@ public static PeerChainValidator create(
     return new PeerChainValidator(storageClient, historicalChainData, peer, status);
   }
 
-  public void run() {
+  public CompletableFuture<Boolean> run() {
     if (hasRun.compareAndSet(false, true)) {
-      executeCheck();
+      result = executeCheck();
     }
+    return result;
```

Comment:
Why not just return the result? that way it would be more inline with how we use completable future's in syncManager and peerSync.

Reply:
For now, we only want to run this once, and this makes `run()` idempotent.  Don't think it matters too much at this point - we can always adjust later.